### PR TITLE
WebRx Implementation - All tests passing

### PR DESCRIPTION
**Pull request by oliverw**
##  WebRx

This is the TodoMVC implementation using [WebRx](http://webrxjs.org).

WebRx is a Javascript MVVM-Framework built on ReactiveX for Javascript (RxJs) that combines functional-reactive programming with observable-powered declarative Data-Binding, Templating and client-side Routing.

An introduction to the framework along with full documentation and code examples can be found at the project [website](http://webrxjs.org).

Feel free to drop me a message [here](https://github.com/oliverw) or on [Twitter](https://twitter.com/webrxjs) if you have any further questions.

## Test results

```javascript
  TodoMVC - webrx
    When page is initially opened
      √ should focus on the todo input field (77ms)
    No Todos
      √ should hide #main and #footer (263ms)
    New Todo
      √ should allow me to add todo items (492ms)
      √ should clear text input field when an item is added (438ms)
      √ should append new items to the bottom of the list (572ms)
      √ should trim text input (363ms)
      √ should show #main and #footer when items added (583ms)
    Mark all as completed
      √ should allow me to mark all items as completed (714ms)
      √ should allow me to clear the completion state of all items (642ms)
      √ complete all checkbox should update state when items are completed / cleared (570ms)
    Item
      √ should allow me to mark items as complete (451ms)
      √ should allow me to un-mark items as complete (441ms)
      √ should allow me to edit an item (746ms)
      √ should show the remove button on hover
    Editing
      √ should hide other controls when editing (644ms)
      √ should save edits on enter (975ms)
      √ should save edits on blur (1353ms)
      √ should trim entered text (1011ms)
      √ should remove the item if an empty text string was entered (771ms)
      √ should cancel edits on escape (1410ms)
    Counter
      √ should display the current number of todo items (535ms)
    Clear completed button
      √ should display the correct text (1037ms)
      √ should remove completed items when clicked (1039ms)
      √ should be hidden when there are no items that are completed (642ms)
    Routing
      √ should allow me to display active items (971ms)
      √ should respect the back button (1057ms)
      √ should allow me to display completed items (720ms)
      √ should allow me to display all items (844ms)
      √ should highlight the currently applied filter (779ms)


  29 passing (1m)
```

### CODE REVIEW

**Comment by arthurvr** [comment]
Thanks for your pull request! We'll discuss this and come back at you.

**Comment by samccone** [comment]
@oliverw mind squashing this into a single commit

@PascalPrecht do you want to help reviewing this?

**Comment by PascalPrecht** [comment]
@samccone sure you didn't mean @passy ?

**Comment by samccone** [comment]
@mattpodwysocki did you want to take a look at this also?

**Comment by samccone** [comment]
@passy  :white_check_mark:

**Comment by oliverw** [comment]
@samccone Squashed.

**Comment by samccone** [comment]
@oliverw looks like you did not push your branch :)

**Comment by oliverw** [comment]
Yeah I've noticed that too. Somehow git rebase -i master isn't working as expected.

**Comment by oliverw** [comment]
Ok fixed

**Comment by passy** [comment]
Yes, will review this later! :)

File: examples/webrx/.gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+node_modules
+.ntvs_analysis.dat
```
Comment (passy) [inline examples/webrx/.gitignore]: Could you limit this to build and npm artifacts?

File: examples/webrx/Gruntfile.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+            }
+        }
+    };
+    
+    grunt.initConfig(conf);
+
+    grunt.loadNpmTasks("grunt-contrib-connect");
+
+    grunt.registerTask("default", ["connect"]);
+};
```
Comment (passy) [inline examples/webrx/Gruntfile.js]: I don't think this is needed.

File: examples/webrx/index.html [hunk 3]
```diff
@@ -0,0 +1,79 @@
+﻿<!DOCTYPE html>
+
+<html data-framework="webrx" lang="en">
+<head>
+    <meta charset="utf-8">
+    <title>WebRx - TodoMVC</title>
```
Comment (passy) [inline examples/webrx/index.html]: `WebRx • TodoMVC`

File: examples/webrx/index.html [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    <link href="node_modules/todomvc-common/base.css" rel="stylesheet">
+    <link href="node_modules/todomvc-app-css/index.css" rel="stylesheet">
+</head>
+
+<body>
+    <section id="todoapp">
+        <header id="header">
+            <h1>todos</h1>
+            <input autofocus="" data-bind="textInput: @current,
```
Comment (passy) [inline examples/webrx/index.html]: No need for `=""` with empty attributes. Or is that needed by the framework somehow?

File: examples/webrx/index.html [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    </section>
+
+    <footer id="info">
+        <p>Double-click to edit a todo</p>
+        <p>Written by <a href="http://webrxjs.org">Oliver Weichhold</a></p>
+        <p>Part of <a href="http://todomvc.com">TodoMVC</a></p>
+    </footer>
+    
+    <script src="node_modules/todomvc-common/base.js"></script> 
+    <script src="node_modules/rx/dist/rx.all.js">
```
Comment (passy) [inline examples/webrx/index.html]: `</script>` on the same line, please.

File: examples/webrx/js/app.js [hunk 6]
```diff
@@ -0,0 +1,156 @@
+/*global wx */
+(function () {
+	'use strict';
```
Comment (passy) [inline examples/webrx/js/app.js]: Indentation.

File: examples/webrx/js/app.js [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+    
+    var localStorageKey = 'todos-webrx';
```
Comment (passy) [inline examples/webrx/js/app.js]: Tabs for indent please.

File: examples/webrx/js/app.js [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+    var displayModeAll = "all";
+    var displayModeActive = "active";
+    var displayModeCompleted = "completed";
```
Comment (passy) [inline examples/webrx/js/app.js]: Could you run the style checkers on this? There are a few things like double vs single quotes here.

File: examples/webrx/js/app.js [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+        // count of todos that are not complete
+        this.remainingCount = Rx.Observable.merge(this.todos.listChanged, this.todos.itemChanged)
+            .select(this.countRemaining, this)
+            .toProperty(this.countRemaining());
+        
+        // setup routing
+        var self = this;
+
+        wx.router.state({
+            name: "$",
```
Comment (passy) [inline examples/webrx/js/app.js]: `function ()`

File: examples/webrx/js/app.js [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+                self.showMode(displayModeActive);
+            }
+        }).state({
+            name: "completed",
+            onEnter: function () {
+                self.showMode(displayModeCompleted);
+            }
+        });
+
+        wx.router.go("$", null, { force: true, location: 2 });
```
Comment (passy) [inline examples/webrx/js/app.js]: Could you add a comment here? The magic string might not be clear.

File: examples/webrx/js/app.js [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+        
+        // persistence
+        Rx.Observable.merge(this.todos.listChanged, this.todos.itemChanged)
+            .throttle(500)
```
Comment (passy) [inline examples/webrx/js/app.js]: Love this!

File: examples/webrx/package.json [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "Oliver Weichhold",
+  "license": "ISC",
+  "devDependencies": {
+    "grunt": "^0.4.5",
+    "grunt-contrib-connect": "^0.9.0"
+  },
+  "dependencies": {
+    "rx": "~2.5.2",
```
Comment (passy) [inline examples/webrx/package.json]: `^` specifier if possible please

File: examples/webrx/.gitignore [hunk 13]
```diff
@@ -0,0 +1,4 @@
...
+obj
+bin
```
Comment (arthurvr) [inline examples/webrx/.gitignore]: also add a final newline

File: examples/webrx/index.html [hunk 14]
```diff
@@ -0,0 +1,79 @@
+﻿<!DOCTYPE html>
+
```
Comment (arthurvr) [inline examples/webrx/index.html]: remove this line.

File: examples/webrx/index.html [hunk 15]
```diff
@@ -0,0 +1,79 @@
+﻿<!DOCTYPE html>
```
Comment (arthurvr) [inline examples/webrx/index.html]: use a lowercase doctype for consistency.

File: examples/webrx/index.html [hunk 16]
```diff
@@ -0,0 +1,79 @@
...
+
+<html data-framework="webrx" lang="en">
+<head>
```
Comment (arthurvr) [inline examples/webrx/index.html]: Can you please indent inside of the `<html>` as well.

File: examples/webrx/index.html [hunk 17]
```diff
@@ -0,0 +1,79 @@
...
+        <p>Written by <a href="http://webrxjs.org">Oliver Weichhold</a></p>
+        <p>Part of <a href="http://todomvc.com">TodoMVC</a></p>
+    </footer>
+    
+    <script src="node_modules/todomvc-common/base.js"></script> 
+    <script src="node_modules/rx/dist/rx.all.js">
+    </script> <script src="node_modules/webrx/dist/web.rx.js"></script> 
+    <script src="js/app.js"></script>
+</body>
+</html>
```
Comment (arthurvr) [inline examples/webrx/index.html]: missing newline

File: examples/webrx/js/app.js (hunk 7 above)
Comment (arthurvr) [inline examples/webrx/js/app.js]: :+1: same applies to all other files.

File: examples/webrx/js/app.js [hunk 18]
```diff
@@ -0,0 +1,156 @@
...
+            this.todos.forEach(function (todo) {
+                todo.completed(this.remainingCount());
+            }, this);
+        }, this);
+        
+        // remove all completed entries
+        this.removeCompletedCmd = wx.command(function () {
+            this.todos.filter(function (todo) {
+                return todo.completed();
+            }).forEach(function(item) {
```
Comment (arthurvr) [inline examples/webrx/js/app.js]: This line but also all the other lines where you've done it `function(a, b)` -> `function (a, b)` please ;)

File: examples/webrx/js/app.js (hunk 9 above)
Comment (arthurvr) [inline examples/webrx/js/app.js]: also, single quotes please.

File: examples/webrx/package.json [hunk 19]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "webrx-todomvc",
+  "version": "1.0.0",
+  "description": "TodoMVC implementation using the WebRx Framework",
+  "main": "js/app.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (arthurvr) [inline examples/webrx/package.json]: can we remove this? (the whole `scripts` in fact)

Comment (arthurvr) [inline examples/webrx/package.json]: and all other moot properties as well. Only needs `private` and all deps.

File: examples/webrx/readme.md [hunk 20]
```diff
@@ -0,0 +1,14 @@
...
+> _[WebRx - webrxjs.org](http://webrxjs.org/)_
+
+## Learning WebRx
+
+The [WebRx website](http://webrxjs.org) is a great resource for getting started.
+
+Here are some links you may find helpful:
+
+* [Getting Started](http://webrxjs.org/docs/)
+* [Browser Compatibility](http://webrxjs.org/docs/browser-support.html)
```
Comment (arthurvr) [inline examples/webrx/readme.md]: can't we come up with some more?

**Comment by arthurvr** [comment]
Mind running jscs and jshint against your app? At some places the code style is wrong..

**Comment by arthurvr** [comment]
You didn't add the framework to the home page yet.

File: examples/webrx/.gitignore [hunk 21]
```diff
@@ -0,0 +1,4 @@
+node_modules
```
Comment (arthurvr) [inline examples/webrx/.gitignore]: by the way this is not true. We need all front-end deps in the repo, could you add them? Similar to the other examples. It's needed for the site to run fine on the live site.

File: examples/webrx/index.html [hunk 22]
```diff
@@ -0,0 +1,79 @@
...
+                </li>
+            </ul>
+            
+            <button data-bind="visible: completedCount, command: removeCompletedCmd" id="clear-completed">Clear completed</button>
+        </footer>
+    </section>
+
+    <footer id="info">
+        <p>Double-click to edit a todo</p>
+        <p>Written by <a href="http://webrxjs.org">Oliver Weichhold</a></p>
```
Comment (arthurvr) [inline examples/webrx/index.html]: this needs to link to a personal website/github profile or whatever you prefer. Not the framework itself.

**Comment by arthurvr** [comment]
As seen in all other apps please include the deps in the repo (also build up a proper [gitignore](https://github.com/tastejs/todomvc/blob/master/examples/backbone_marionette/.gitignore))

File: examples/webrx/readme.md (hunk 20 above)
Comment (arthurvr) [inline examples/webrx/readme.md]: some more links but also more info regarding implementation maybe? (check other readmes)

File: examples/webrx/package.json (hunk 12 above)
Comment (oliverw) [inline examples/webrx/package.json]: Not possible right now. Even the 2.5 introduced some breaking changes.

File: examples/webrx/readme.md (hunk 20 above)
Comment (oliverw) [inline examples/webrx/readme.md]: Sure. Working on it.

**Comment by mattpodwysocki** [comment]
@oliverw which parts of 2.5 broke WebRx for you?  The only thing that changed was moving the rx-node functionality elsewhere due to issues with React.

**Comment by mattpodwysocki** [comment]
@samccone yeah, I was the one who prompted him to submit it since it looked pretty solid...

**Comment by oliverw** [comment]
// as of RxJs 2.5.2 this is missing in rx.all.d.ts
export interface SchedulerStatic {
   isScheduler(o: any): boolean;
}

Formerly known as Rx.helpers.isScheduler()

**Comment by mattpodwysocki** [comment]
@oliverw oh crap, I was hoping that wasn't going to happen. Sorry, will fix...

**Comment by oliverw** [comment]
@mattpodwysocki No problem.

**Comment by oliverw** [comment]
Allright, I think all of the issues have been fixed/added.

**Comment by arthurvr** [comment]
Why do we need a gruntfile? Seems like it doesn't have enough logic. A server should be runned from the root directory anyways.

File: examples/webrx/index.html [hunk 23]
```diff
@@ -0,0 +1,78 @@
...
+
+		<footer id="info">
+			<p>Double-click to edit a todo</p>
+			<p>Written by <a href="https://github.com/WebRxJS/WebRx">Oliver Weichhold</a></p>
+			<p>Part of <a href="http://todomvc.com">TodoMVC</a></p>
+		</footer>
+		
+		<script src="node_modules/todomvc-common/base.js"></script> 
+		<script src="node_modules/rx/dist/rx.all.js"></script>
+        <script src="node_modules/webrx/dist/web.rx.js"></script>
```
Comment (arthurvr) [inline examples/webrx/index.html]: this indentation is wrong.

File: examples/webrx/index.html [hunk 24]
```diff
@@ -0,0 +1,78 @@
+﻿<!doctype html>
+<html data-framework="webrx" lang="en">
+	<head>
+		<meta charset="utf-8">
+		<title>WebRx - TodoMVC</title>
+		<link href="node_modules/todomvc-common/base.css" rel="stylesheet">
+		<link href="node_modules/todomvc-app-css/index.css" rel="stylesheet">
+	</head>
+
```
Comment (arthurvr) [inline examples/webrx/index.html]: remove this line :)

File: examples/webrx/index.html [hunk 25]
```diff
@@ -0,0 +1,78 @@
...
+					</li>
+				</ul>
+				
+				<button data-bind="visible: completedCount, command: removeCompletedCmd" id="clear-completed">Clear completed</button>
+			</footer>
+		</section>
+
+		<footer id="info">
+			<p>Double-click to edit a todo</p>
+			<p>Written by <a href="https://github.com/WebRxJS/WebRx">Oliver Weichhold</a></p>
```
Comment (arthurvr) [inline examples/webrx/index.html]: This now links to a github repo. It should be the site/profile of the author of the app.

File: examples/webrx/js/app.js [hunk 26]
```diff
@@ -0,0 +1,162 @@
+/*global wx */
+(function () {
+    'use strict';
```
Comment (arthurvr) [inline examples/webrx/js/app.js]: wrong indentation

File: examples/webrx/.gitignore [hunk 27]
```diff
@@ -0,0 +1,2 @@
+obj
+bin
```
Comment (arthurvr) [inline examples/webrx/.gitignore]: please take a look at other gitignores. it should include all module files we don't need (and those should be removed from the commit)

**Comment by arthurvr** [comment]
> You didn't add the framework to the home page yet.

As a reminder, this still needs to happen.

**Comment by arthurvr** [comment]
Also doesn't have a learn.json entry yet.

File: examples/webrx/js/app.js [hunk 28]
```diff
@@ -0,0 +1,162 @@
...
+			if (!trimmedTitle) {
+				this.todos.remove(item);
+			}
+		}, this);
+
+		this.countCompleted = function () {
+			return this.todos.filter(function (todo) {
+				return todo.completed();
+			}).length;
+		}
```
Comment (arthurvr) [inline examples/webrx/js/app.js]: Can you please run jshint as we asked before? This misses a semicolon as well as line 116.

File: examples/webrx/js/app.js [hunk 29]
```diff
@@ -0,0 +1,162 @@
...
+		// persistence
+		Rx.Observable.merge(this.todos.listChanged, this.todos.itemChanged)
+			.throttle(500)
+			.subscribeOnNext(function () {
+				localStorage.setItem(localStorageKey, JSON.stringify(this.todos.map(function (x) {
+					return { title: x.title(), completed: x.completed() };
+			})));
+		}, this);
+	};	
+
```
Comment (arthurvr) [inline examples/webrx/js/app.js]: trailing whitespace. Can you please run jscs?

**Comment by arthurvr** [comment]
Our comments on package.json weren't addressed yet.

**Comment by oliverw** [comment]
> You didn't add the framework to the home page yet.

How?

> Also doesn't have a learn.json entry yet.

See above :)

**Comment by arthurvr** [comment]
> You didn't add the framework to the home page yet.

Add it after https://github.com/tastejs/todomvc/blob/master/index.html#L277

> Also doesn't have a learn.json entry yet.

You can [look at the other apps as example](https://github.com/tastejs/todomvc/blob/master/learn.json) ;)

**Comment by oliverw** [comment]
> and all other moot properties as well. Only needs private and all deps.

"private"?

**Comment by arthurvr** [comment]
> "private"?

See [the npm docs](https://docs.npmjs.com/files/package.json#private)

**Comment by oliverw** [comment]
- package.json updated
- missing semicolons fixed
- profile link

**Comment by passy** [comment]
@oliverw I think you've commited `node_modules` entirely or something. Can't open the diff on my tablet anymore.

**Comment by arthurvr** [comment]
Yup. As I said you'll need to have a proper [gitignore](https://github.com/tastejs/todomvc/blob/master/examples/backbone_marionette/.gitignore).

**Comment by oliverw** [comment]
@passy Yeah I did. I was very surprised when someone here requested that the dependencies should be included. Originally node_modules was part of .gitignore.

**Comment by oliverw** [comment]
+learn.json

**Comment by oliverw** [comment]
Done.

**Comment by arthurvr** [comment]
Thank you, @oliverw! I'll go over this once again this evening.

File: examples/webrx/.gitignore [hunk 30]
```diff
@@ -0,0 +1,19 @@
...
+node_modules/webrx/dist/*
+!node_modules/webrx/dist
+!node_modules/webrx/dist/web.rx.js
+
+node_modules/todomvc-app-css/*
+!node_modules/todomvc-app-css/index.css
+
+node_modules/todomvc-common/*
+!node_modules/todomvc-common/base.css
+!node_modules/todomvc-common/base.js
```
Comment (arthurvr) [inline examples/webrx/.gitignore]: missing newline

File: examples/webrx/Gruntfile.js [hunk 31]
```diff
@@ -0,0 +1,20 @@
...
+                }
+            }
+        }
+    };
+    
+    grunt.initConfig(conf);
+
+    grunt.loadNpmTasks("grunt-contrib-connect");
+
+    grunt.registerTask("default", ["connect"]);
```
Comment (arthurvr) [inline examples/webrx/Gruntfile.js]: the indentation in this file is wrong. We use tabs, this uses 4 spaces.

File: examples/webrx/readme.md [hunk 32]
```diff
@@ -0,0 +1,26 @@
...
+- [Commands](http://webrxjs.org/docs/commands.html)
+- [Data-Binding](http://webrxjs.org/docs/binding-syntax.html)
+- [Routing](http://webrxjs.org/docs/routing-overview.html)
+
+## Learning WebRx
+
+Here are some links you may find helpful:
+
+* [Website](http://webrxjs.org) is a great resource for getting started.
+* [Documentation](http://webrxjs.org/docs/)
```
Comment (arthurvr) [inline examples/webrx/readme.md]: Can we try to format this file as the other readmes? (The quote on the top and the overall structure. The rest looks awesome.) A good example is [the angular one](https://raw.githubusercontent.com/tastejs/todomvc/master/examples/angularjs/readme.md).

File: index.html [hunk 33]
```diff
@@ -129,6 +129,9 @@
 								<li class="routing">
 									<a href="examples/troopjs_require/" data-source="https://github.com/troopjs/" data-content="TroopJS attempts to package popular front-end technologies and bind them with minimal effort for the developer. It includes jQuery for DOM manipulation, When.js for promises, RequireJS for modularity and Has.js for feature detection. On top, it includes Pub/Sub support, templating, weaving (widgets to DOM) and auto-wiring.">TroopJS + RequireJS</a>
 								</li>
+								<li class="routing">
+									<a href="examples/webrx/" data-source="http://webrxjs.org" data-content="WebRx is a Javascript MVVM-Framework that combines functional-reactive programming with declarative Data-Binding, Templating and Client-Side Routing. The framework is built on top of ReactiveX for Javascript (RxJs) which is a powerful set of libraries for processing and querying asynchronous data-streams that can originate from diverse sources such as Http-Requests, Input-Events, Timers and much more.">WebRx</a>
+								</li>
```
Comment (arthurvr) [inline index.html]: please add it to labs first ;)

**Comment by arthurvr** [comment]
Tests pass. Except those two comments this LGTM, thus. Awesome work! @passy @sindresorhus Would love it if you take a look as well.

**Comment by samccone** [comment]
Looks like quite a few jscs style related errors hanging around still

https://gist.github.com/samccone/cb7a5631997ccb9d7761

**Comment by oliverw** [comment]
Done.

**Comment by samccone** [comment]
Looks like you are just about there @oliverw :)

```
> @ lint /Users/sam/Desktop/repos/todomvc/tooling
> jscs -c ../.jscsrc ../examples/webrx/

Expected indentation of 1 characters at ../examples/webrx/Gruntfile.js :
     1 |module.exports = function (grunt) {
     2 |    "use strict";
---------^
     3 |
     4 |    var conf = {

Invalid quote mark found at ../examples/webrx/Gruntfile.js :
     1 |module.exports = function (grunt) {
     2 |    "use strict";
------------^
     3 |
     4 |    var conf = {

Expected indentation of 1 characters at ../examples/webrx/Gruntfile.js :
     2 |    "use strict";
     3 |
     4 |    var conf = {
---------^
     5 |        connect: {
     6 |            server: {

Expected indentation of 1 characters at ../examples/webrx/Gruntfile.js :
    13 |    };
    14 |
    15 |    grunt.initConfig(conf);
---------^
    16 |
    17 |    grunt.loadNpmTasks("grunt-contrib-connect");

Expected indentation of 1 characters at ../examples/webrx/Gruntfile.js :
    15 |    grunt.initConfig(conf);
    16 |
    17 |    grunt.loadNpmTasks("grunt-contrib-connect");
---------^
    18 |
    19 |    grunt.registerTask("default", ["connect"]);

Invalid quote mark found at ../examples/webrx/Gruntfile.js :
    15 |    grunt.initConfig(conf);
    16 |
    17 |    grunt.loadNpmTasks("grunt-contrib-connect");
-------------------------------^
    18 |
    19 |    grunt.registerTask("default", ["connect"]);

Expected indentation of 1 characters at ../examples/webrx/Gruntfile.js :
    17 |    grunt.loadNpmTasks("grunt-contrib-connect");
    18 |
    19 |    grunt.registerTask("default", ["connect"]);
---------^
    20 |}
    21 |

Invalid quote mark found at ../examples/webrx/Gruntfile.js :
    17 |    grunt.loadNpmTasks("grunt-contrib-connect");
    18 |
    19 |    grunt.registerTask("default", ["connect"]);
-------------------------------^
    20 |}
    21 |

Invalid quote mark found at ../examples/webrx/Gruntfile.js :
    17 |    grunt.loadNpmTasks("grunt-contrib-connect");
    18 |
    19 |    grunt.registerTask("default", ["connect"]);
-------------------------------------------^
    20 |}
    21 |

Illegal trailing whitespace at ../examples/webrx/js/app.js :
     2 |(function () {
     3 | 'use strict';
     4 | 
---------^
     5 | var localStorageKey = 'todos-webrx';
     6 | var displayModeAll = 'all';

Illegal trailing whitespace at ../examples/webrx/js/app.js :
    23 |
    24 |  // we want to get notified of changes to any of the todos contained in the list
    25 |  // not just of structural changes to the list (via "listChanged" obserable). 
---------------------------------------------------------------------------------------^
    26 |  // Those changes are then exposed using the list's "itemChanged" obseravable
    27 |  this.todos.changeTrackingEnabled = true;

Illegal trailing whitespace at ../examples/webrx/js/app.js :
    27 |  this.todos.changeTrackingEnabled = true;
    28 |
    29 |  this.current = wx.property();  
-----------------------------------------^
    30 |  this.showMode = wx.property('all');
    31 |

Illegal trailing whitespace at ../examples/webrx/js/app.js :
   111 |   .select(this.countCompleted, this)
   112 |   .toProperty(this.countCompleted());
   113 |  
----------^
   114 |  this.countRemaining = function () {
   115 |   return this.todos.length() - this.completedCount();

Expected indentation of 4 characters at ../examples/webrx/js/app.js :
   150 |    localStorage.setItem(localStorageKey, JSON.stringify(this.todos.map(function (x) {
   151 |     return { title: x.title(), completed: x.completed() };
   152 |   })));
------------^
   153 |  }, this);
   154 | }; 

Expected indentation of 3 characters at ../examples/webrx/js/app.js :
   151 |     return { title: x.title(), completed: x.completed() };
   152 |   })));
   153 |  }, this);
-----------^
   154 | }; 
   155 |

Illegal trailing whitespace at ../examples/webrx/js/app.js :
   152 |   })));
   153 |  }, this);
   154 | }; 
------------^
   155 |
   156 | // check local storage for todos


16 code style errors found.
```

**Comment by oliverw** [comment]
Thank you Sublime :+1:

**Comment by samccone** [comment]
down to 10!
https://gist.github.com/samccone/b22a610e881ceb2be7d4

**Comment by oliverw** [comment]
Check again :)

**Comment by samccone** [comment]
only 4 left :smile: 

https://gist.github.com/samccone/8e540fc736a50a44f0ff

**Comment by oliverw** [comment]
Ahhhh

**Comment by oliverw** [comment]
:+1:

**Comment by samccone** [comment]
:clap: all passing

**Comment by passy** [comment]
Thanks for fixing all the things! :)

One thing I've noticed locally is that the demo rewrites the routes really aggressively. Instead of showing "http://localhost:8080/examples/webrx/" is only shows "http://localhost:8080/" and even seems to use replaceState so I can't navigate back. I haven't looked into the code for that, but is anyone else seeing this?

**Comment by oliverw** [comment]
@passy Good catch. Looking into it.

**Comment by oliverw** [comment]
@passy The router is now configured with a baseUrl that is supposed to correspond the TodoMVC production setup.

**Comment by passy** [comment]
@oliverw Could you configure it to use hash-bangs instead of full URLs? We don't support custom routes at todomvc.com so the deep-links are currently broken if you reload the page after selecting a filter.

There were some remaining white-space and indentation issues. If you could just apply that patch here:
https://gist.github.com/passy/4a9a19b79ba62538b24e

The rest looks fine! Happy to merge this after those two things are addressed. :)

**Comment by oliverw** [comment]
@passy Sorry the WebRx was built for modern browsers and does not support hash-bang urls. ;(

> We don't support custom routes at todomvc.com so the deep-links are currently broken if you reload the page after selecting a filter.

What something like this be an option?:

http://webrxjs.org/docs/routing-overview.html (scroll down to "Server side setup")

**Comment by passy** [comment]
@oliverw Server side configuration is not an option, unfortunately. One of the requirements for all apps is that they run from an extracted tarball on anyone's system without build step or server setup.

**Comment by oliverw** [comment]
@passy Ah ok. Well it should still work - except for deep-linking.

**Comment by passy** [comment]
@oliverw We discussed that and we're happy to leave it like this if we document it and open an issue about it. Could you stick a note into the readme about this? If you then apply the whitespace patch, we should see this baby in master in no time! :)

File: examples/webrx/index.html (hunk 23 above)
Comment (arthurvr) [inline examples/webrx/index.html]: ping. This needs fixing as well.

**Comment by oliverw** [comment]
@passy Applied :)

**Comment by passy** [comment]
Landed! :space_invader: 



Thanks @oliverw and @arthurvr for your hard work!

**Comment by oliverw** [comment]
Haha Cool. :+1:

**Comment by arthurvr** [comment]
I did two other small fixes on master, no big deals. Neat work @oliverw!

**Comment by samccone** [comment]
:+1: great work @oliverw thanks for putting up with the review :package:

**Comment by oliverw** [comment]
@samccone I actually enjoyed working with you guys. :)

**Pull request merged**